File: src/services/DebugService.cpp
#include "DebugService.hpp"
#include <print>

DebugService::DebugService(HyprlandListener& listener) : listener(listener) {
    this->subscription_id = this->listener.subscribe([this](const HyprlandEvent& event) { this->onEvent(event); });
}

DebugService::~DebugService() {
    this->listener.unsubscribe(this->subscription_id);
}

void DebugService::onEvent(const HyprlandEvent& event) {
    // This service's only job is to log everything.
    std::println("[DEBUG] Event: {0} | Data: {1}", event.type, event.data);
}

File: src/services/debug_service.rs
use std::sync::{Arc, Mutex};

use crate::hyprland::hyprland_ipc::HyprlandIpc;

pub struct DebugService;

impl DebugService {
  pub fn new(listener: Arc<Mutex<HyprlandIpc>>) -> Self {
    if let Ok(mut guard) = listener.lock() {
      guard.subscribe(Box::new(|event| {
        println!("[Debug] Event Received: {event:?}");
      }));
      println!("[DebugService] Subscribed to listener.");
    }

    Self
  }
}
